src/solvers/workListLocTrace.ml

```diff
@@ -0,0 +1,111 @@
+open Batteries
+open GobPretty
+open Analyses
+open Constraints
+open PostSolvingFlag
+open PriorityCalc
+
+module Make =
+  functor (S:EqConstrSys) ->
+  functor (HM:Hashtbl.S with type key = S.v) ->
+  struct
+
+    include Generic.SolverStats (S) (HM)
+
+    module ListQueue = struct
+      type 'a t = 'a list
+      exception Empty
+      let empty = []
+      let is_empty = function [] -> true | _ -> false
+      let to_list = Fun.id
+      let from_list = Fun.id
+      let enqueue st elem =
+        let stList = to_list st in
+        if List.mem elem stList then (
+          st
+        )
+        else
+          List.append st [elem]
+      let peek = function [] -> raise Empty | x :: _ -> x
+      let pop = function [] -> raise Empty | x :: s -> x, s
+      let size = List.length
+
+      let fold f a b = List.fold_left f b (to_list a)
+
+    end
```

Comment:
Why not just use the standard `Queue`? That should work just fine with a worklist solver.
The current implementation is quite inefficient since it's appending to the end of the list.

Or if functional queues are needed, there's `BatDeque`, which implements an efficient functional one.